third_party/move/move-model/src/ty.rs

```diff
@@ -122,17 +128,224 @@ pub enum Constraint {
     /// The type variable must be instantiated with a type for which a receiver function with the given
     /// signature exists: the name, the optional type arguments, the argument types, and the
     /// result type.
-    SomeReceiverFunction(Symbol, Option<Vec<Type>>, Vec<(Loc, Type)>, Type),
-    /// The type variable must be instantiated with a type which has the given ability
-    HasAbility(Ability),
+    SomeReceiverFunction(
+        Symbol,
+        /// The optional type arguments, with locations
+        Option<(Vec<Loc>, Vec<Type>)>,
+        /// The locations of the arguments
+        Vec<Loc>,
+        /// The argument type
+        Vec<Type>,
+        /// The result type
+        Type,
+    ),
+    /// The type must not be reference because it is used as the type of some field or
+    /// as a type argument.
+    NoReference,
```

Comment:
Not sure why this would be clearer? I thought `NoReference` is the right English here, compared to `NotReference` which is not a valid word.

Reply:
The constraint is that it is "not a reference" => `NotReference`.

Both `NoReference` and `NotReference` are not valid words by themselves.

Anyway, I have marked this as a nit because the comment makes the constraint clear to me.